radicle/src/identity.rs

```diff
@@ -101,50 +98,45 @@ impl Identity<Untrusted> {
 
         // Retrieve root document.
         let root_oid = history.pop().ok_or(IdentityError::MissingRoot)??.into();
-        let root_blob = Doc::blob_at(root_oid, repo)?;
-        let root: git::Oid = root_blob.id().into();
-        let trusted = Doc::from_json(root_blob.content())?;
+        let root = Doc::<Verified>::load_at(root_oid, repo)?;
         let revision = history.len() as u32;
 
-        let mut trusted = trusted.verified()?;
-        let mut current = root;
-        let mut signatures = Vec::new();
+        // Every identity founder must have signed the root document.
+        for founder in &root.doc.delegates {
+            if !root.sigs.iter().any(|(k, _)| k == &**founder) {
+                return Err(IdentityError::MissingRootSignatures);
+            }
+        }
```

Comment:
I believe you can do something like this:
```suggestion
        if !root.doc.delegates.all(|founder| root.sigs.contains_key(founder)) {
            return Err(IdentityError::MissingRootSignatures)
        }
```

Reply:
Ah, you would need to change `sigs` to a `HashMap`